vault/token_store.go

```diff
@@ -73,12 +73,27 @@ var (
 	// pathSuffixSanitize is used to ensure a path suffix in a role is valid.
 	pathSuffixSanitize = regexp.MustCompile("\\w[\\w-.]+\\w")
 
-	destroyCubbyhole = func(ctx context.Context, ts *TokenStore, saltedID string) error {
+	destroyCubbyhole = func(ctx context.Context, ts *TokenStore, te *TokenEntry) error {
 		if ts.cubbyholeBackend == nil {
 			// Should only ever happen in testing
 			return nil
 		}
-		return ts.cubbyholeBackend.revoke(ctx, salt.SaltID(ts.cubbyholeBackend.saltUUID, saltedID, salt.SHA1Hash))
+
+		switch {
+		case te.Version < 2:
+			// This is only here for backwards compatibility
+			saltedID, err := ts.SaltID(ctx, te.ID)
```

Comment:
Should we add a "Deprecated:" comment to the `TokenStore` `SaltID` method? In my code editor, that'll cause there to be a strike-through on the method if I ever try to use it. So it's still there but I get a strong indication I shouldn't be newly using it. Not sure if that's appropriate for that salting method, and/or any of the others we're moving away from.

Reply:
Done.